lib/migrate/extractTextPlugin/extractTextPlugin.js

```diff
@@ -1,58 +1,111 @@
-const utils = require("../../utils/ast-utils");
+const {
+	addRequire,
+	createPluginByName,
+	removePluginByName,
+	getLoaderOptionValue,
+	getPluginOptionValue,
+	isType,
+	pathToArray,
+	removeRequire,
+	safeTraverse
+} = require("../../utils/ast-utils");
 
 /**
  *
- * Check whether `node` is the invocation of the plugin denoted by `pluginName`
- *
- * @param {Object} j - jscodeshift top-level import
- * @param {Node} node - ast node to check
- * @param {String} pluginName - name of the plugin
- * @returns {Boolean} isPluginInvocation - whether `node` is the invocation of the plugin denoted by `pluginName`
- */
-
-function findInvocation(j, node, pluginName) {
-	let found = false;
-	found =
-		j(node)
-			.find(j.MemberExpression)
-			.filter(p => p.get("object").value.name === pluginName)
-			.size() > 0;
-	return found;
-}
-
-/**
- *
- * Transform for ExtractTextPlugin arguments. Consolidates arguments into single options object.
+ * Transform for ExtractTextPlugin. Replaces it with MiniCssExtractPlugin and translates
+ * configuration where possible.
  *
  * @param {Object} j - jscodeshift top-level import
  * @param {Node} ast - jscodeshift ast to transform
  * @returns {Node} ast - jscodeshift ast
  */
+module.exports = function transformExtractTextPlugin(j, ast) {
+	const devLoaders = [];
+	const prodLoaders = [];
+
+	let elements;
+	let loaderIndex;
+	let loaderPath;
+	let newElements;
+	let pluginExists;
+	let requireExists;
+
+	const filename = getPluginOptionValue(j, ast, "ExtractTextPlugin", "filename");
+	const publicPath = getLoaderOptionValue(j, ast, "ExtractTextPlugin", "publicPath");
+	const currentFallbackLoader = getLoaderOptionValue(j, ast, "ExtractTextPlugin", "fallback");
+	const currentLoader = getLoaderOptionValue(j, ast, "ExtractTextPlugin", "use");
 
-module.exports = function(j, ast) {
-	const changeArguments = function(p) {
-		const args = p.value.arguments;
-
-		const literalArgs = args.filter(p => utils.isType(p, "Literal"));
-		if (literalArgs && literalArgs.length > 1) {
-			const newArgs = j.objectExpression(
-				literalArgs.map((p, index) =>
-					utils.createProperty(j, index === 0 ? "fallback" : "use", p.value)
-				)
-			);
-			p.value.arguments = [newArgs];
+	const loaderOptions = filename || publicPath ? {} : null;
+	filename && Object.assign(loaderOptions, { filename });
+	publicPath && Object.assign(loaderOptions, { publicPath: publicPath.value });
+
+	// Remove require for old plugin
+	requireExists = removeRequire(j, ast, "ExtractTextPlugin");
+
+	// Add require for new plugin
+	requireExists && addRequire(j, ast, "MiniCssExtractPlugin", "mini-css-extract-plugin");
+
+	// Remove old plugin
+	pluginExists = removePluginByName(j, ast, "ExtractTextPlugin");
+
+	// Add new plugin
+	pluginExists && createPluginByName(j, ast, "MiniCssExtractPlugin", loaderOptions);
+
+	// Remove old loader
+	ast.find(j.Identifier, { name: "ExtractTextPlugin" })
```

Comment:
This could be abstracted into an util, maybe you can make it general so that we can reuse it for later?